lib/phases/cold-launch.js

```diff
@@ -179,6 +180,29 @@ ColdLaunch.prototype.prime = function() {
     });
 };
 
+ColdLaunch.prototype.maybeClearBuffer = function() {
+  if (this.launchesBuffered < Phase.PERFORMANCE_BUFFER_SAFETY) {
+    return Promise.resolve();
+  }
+
+  var phase = this;
+  var marionette = this.device.marionette;
+
+  // Clear the performance buffer after a controlled number of app launches
+  // to keep marking running smoothly, and to avoid hitting the maximum
+  // buffer limit
+  return marionette
+    .startSession()
```

Comment:
Just an idea... maybe instead of starting a new marionette session, have this done during the existing marionette session just before an app launch? Might be better to start and stop as few marionette sessions as possible. However, I realize this wont actually be done that often as it's probably rare to go over 50 app launches.